pypit/arproc.py

```diff
@@ -1103,64 +1116,69 @@ def rn_frame(slf, det):
     rn_img : ndarray
       Read noise *variance* image (i.e. RN**2)
     """
+    dnum = 'det{0:02d}'.format(det)
+
     # Loop on amplifiers
-    rnimg = np.zeros_like(slf._ampsec[det-1])
-    for ii in range(slf._spect['det'][det-1]['numamplifiers']):
+    rnimg = np.zeros_like(slf._datasec[det-1])
+    for ii in range(settings.spect[dnum]['numamplifiers']):
         amp = ii+1
-        amppix = slf._ampsec[det-1] == amp
-        rnimg[amppix] = (slf._spect['det'][det-1]['ronoise'][ii]**2 +
-                         (0.5*slf._spect['det'][det-1]['gain'][ii])**2)
+        amppix = slf._datasec[det-1] == amp
+        rnimg[amppix] = (settings.spect[dnum]['ronoise'][ii]**2 +
+                         (0.5*settings.spect[dnum]['gain'][ii])**2)
     # Return
     return rnimg
 
 
-def sub_overscan(slf, det, file, use_datasec=False):
+def sub_overscan(frame, det, use_datasec=False):
     """
     Subtract overscan
     use_datasec : bool, optional
       Overscan region is limited to datasec, not ampsec
     """
+    dnum = 'det{0:02d}'.format(det)
 
-    for i in range(slf._spect['det'][det-1]['numamplifiers']):
+    for i in range(settings.spect[dnum]['numamplifiers']):
         # Determine the section of the chip that contains the overscan region
         oscansec = "oscansec{0:02d}".format(i+1)
-        ox0, ox1, oy0, oy1 = slf._spect['det'][det-1][oscansec][0][0], slf._spect['det'][det-1][oscansec][0][1], slf._spect['det'][det-1][oscansec][1][0], slf._spect['det'][det-1][oscansec][1][1]
-        if ox0 < 0: ox0 += file.shape[0]
-        if ox1 <= 0: ox1 += file.shape[0]
-        if oy0 < 0: oy0 += file.shape[1]
-        if oy1 <= 0: oy1 += file.shape[1]
+        ox0, ox1 = settings.spect[dnum][oscansec][0][0], settings.spect[dnum][oscansec][0][1]
+        oy0, oy1 = settings.spect[dnum][oscansec][1][0], settings.spect[dnum][oscansec][1][1]
+        if ox0 < 0: ox0 += frame.shape[0]
+        if ox1 <= 0: ox1 += frame.shape[0]
+        if oy0 < 0: oy0 += frame.shape[1]
+        if oy1 <= 0: oy1 += frame.shape[1]
         xos = np.arange(ox0, ox1)
         yos = np.arange(oy0, oy1)
         w = np.ix_(xos, yos)
-        oscan = file[w]
+        oscan = frame[w]
         # Determine the section of the chip that is read out by the amplifier
         ampsec = "ampsec{0:02d}".format(i+1)
-        ax0, ax1, ay0, ay1 = slf._spect['det'][det-1][ampsec][0][0], slf._spect['det'][det-1][ampsec][0][1], slf._spect['det'][det-1][ampsec][1][0], slf._spect['det'][det-1][ampsec][1][1]
-        if ax0 < 0: ax0 += file.shape[0]
-        if ax1 <= 0: ax1 += file.shape[0]
-        if ay0 < 0: ay0 += file.shape[1]
-        if ay1 <= 0: ay1 += file.shape[1]
+        ax0, ax1 = settings.spect[dnum][ampsec][0][0], settings.spect[dnum][ampsec][0][1]
```

Comment:
It will probably become clear to me as I go, but I now see ampsec reappear.  And I wonder how it will be used and set.